### ISSUE TITLE
Handling Exponential Notation in Depth Websocket Payloads

### ISSUE THREAD

**Issue by mchandschuh** [dougdellolio/coinbasepro-csharp#160]
While receiving depth data via websocket, I've recently been experiencing serializing failures due to coinbase sending data in exponential format. The below example pertains to the new coin but I _think_ I experienced this once before the new coin was released. Here's the full content body as intercepted via `JsonConfig.DeserializeObject` where it throws a `JsonReaderException`. For brevity I've trimmed the body down quite a bit:

```
{
  "product_id": "ZRX-BTC",
  "type": "snapshot",
  "bids": [
    [
      "5.1e-7",
      "150000"
    ],
    [
      "5e-7",
      "10000"
    ],
    [
      "1e-7",
      "15288"
    ],
    [
      "5e-8",
      "8000"
    ],
    [
      "3e-8",
      "150000"
    ],
    [
      "2e-8",
      "150000"
    ],
    [
      "1e-8",
      "600000"
    ]
  ],
  "asks": [

  ]
}
```

**Comment by dougdellolio** [comment]
@mchandschuh thanks for reporting this. I am seeing this locally now too. I have sent a ticket to their support team to get more information around this to see if it is a data issue on their end that can be fixed or it is something we should account for here. They usually respond quickly for API support requests so I'll keep you posted. Otherwise, if we don't hear back soon we should go ahead and add the fix in.

**Comment by mchandschuh** [comment]
Here's a commit with the above described changes: https://github.com/1Konto/coinbasepro-csharp/commit/f8fd0ccb46642320f4de78ecd9403cae421c07ba

**Comment by dougdellolio** [comment]
fyi support responded 16 days ago to my request with this:

>  ... I have filed a bug report, and requested our engineering team to investigate further. Thank you for bearing with us in the meantime

I haven't received a response since so I just sent a followup this morning. In the meantime I am going to add your fix in

**Comment by bthtrade** [comment]
Thank you for highlighting this JsonConfig file I've never heard about before.
In the converter, I personally use: 
```
    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
    {
        var value = reader.Value as string;
        if (string.IsNullOrEmpty(value))
        {
            return null;
        }
        return decimal.Parse(value, NumberStyles.Number | NumberStyles.AllowExponent);
    }
```

**Comment by dougdellolio** [comment]
closing. this is fixed with v1.0.25. let me know if you see any other issues
